src/Model/batchprocessing/BatchProcessPyRad2CSV.py

```diff
@@ -0,0 +1,290 @@
+        self.output_path = output_path
+        self.filename = "Pyradiomics_.csv"
+
+    def start(self):
+        """
+        Goes through the steps of the PyRadCSV conversion.
+        """
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        if not self.ready:
+            self.summary = "SKIP"
+            return False
+
+        rtss_path = self.patient_dict_container.filepaths.get('rtss')
+        patient_id = self.patient_dict_container.dataset.get('rtss').PatientID
+        patient_id = BatchProcessPyRad2CSV.clean_patient_id(patient_id)
+        patient_path = self.patient_dict_container.path
+        file_name = \
+            BatchProcessPyRad2CSV.clean_patient_id(patient_id) + '.nrrd'
+        patient_nrrd_folder_path = patient_path + '/nrrd/'
+        patient_nrrd_file_path = patient_nrrd_folder_path + file_name
+
+        output_csv_path = self.output_path.joinpath('CSV')
+
+        # If folder does not exist
+        if not os.path.exists(patient_nrrd_folder_path):
+            # Create folder
+            os.makedirs(patient_nrrd_folder_path)
+
+        # If folder does not exist
+        if not os.path.exists(output_csv_path):
+            # Create folder
+            os.makedirs(output_csv_path)
+
+        self.progress_callback.emit(("Converting dicom to nrrd..", 25))
+
+        # Convert dicom files to nrrd for pyradiomics processing
+        BatchProcessPyRad2CSV.convert_to_nrrd(
+            patient_path, patient_nrrd_file_path)
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        # Location of folder where converted masks saved
+        mask_folder_path = patient_nrrd_folder_path + 'structures'
+        if not os.path.exists(mask_folder_path):
+            os.makedirs(mask_folder_path)
+
+        self.progress_callback.emit(("Converting ROIs to nrrd..", 45))
+
+        # Convert ROIs to nrrd
+        BatchProcessPyRad2CSV.convert_rois_to_nrrd(
+            patient_path, rtss_path, mask_folder_path)
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        self.progress_callback.emit(("Running pyradiomics..", 70))
+
+        # Run pyradiomics, convert to dataframe
+        radiomics_df = BatchProcessPyRad2CSV.get_radiomics_df(
+            patient_path, patient_id, patient_nrrd_file_path, mask_folder_path)
+
+        # Stop loading
+        if self.interrupt_flag.is_set():
+            self.patient_dict_container.clear()
+            self.summary = "INTERRUPT"
+            return False
+
+        if radiomics_df is None:
+            self.summary = "PYRAD_NO_DF"
```

Comment:
If you can make summary messages enum or constants, and import them when you want to use that would be cleaner.